enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,304 @@
+  created for the service.
+- Openshift router should seamlessly reload the OpenShift service CA certificate or the
+  administrator-configured CA certificate bundle when either is modified.
+
+### Non-Goals
+
+- OpenShift administrator defined CA certificate bundle is limited to the IngressController and OpenShift
+  router instance usage and for the routes created with re-encrypt termination type, any other scenario is 
+  outside the scope of this proposal.
+
+## Proposal
+Update IngressController API to:
+1. Make provision for user to provide reference to a configmap containing the trusted CA certificates. 
+
+Change OpenShift Ingress Operator to:
+1. Make the CA certificate bundle defined in `IngressController` available to the OpenShift Router as the 
+   default destination CA bundle file to be used for verifying the services' certificate.
+
+Update router to:
+1. Watch the destination CA certificate bundle and update its content internally when needed.
+
+### Workflow Description
+
+- An OpenShift administrator configures the list of trusted CA certificates as a ConfigMap, and provides 
+  reference to it in the `IngressController` object.
+- The Ingress Operator configures OpenShift router with a ConfigMap volume mount to use the ConfigMap as 
+  the default destination CA certificate to verify a service's certificate when a route with re-encrypt
+  termination type is created with an empty destination CA.
+- OpenShift router watches the volume mount and reloads HAProxy whenever the CA certificate bundle changes.
+
+#### Variation [optional]
+None
+
+### API Extensions
+
+The proposal expects the OpenShift cluster administrator to create a ConfigMap with all the required 
+CA certificates, including the OpenShift service CA in the same namespace of the `IngressController` object
+in which the configmap reference is provided.
+
+Enhancement requires below modifications to the `IngressController` API.
+- Add `defaultDestinationCATrust` field to `.spec` of the `ingresscontrollers.operator.openshift.io`
+```golang
+type IngressControllerSpec struct {
+	// defaultDestinationCATrust is a reference to a configmap containing the trusted
+	// Certificate Authorities (CAs) used by the Router for verifying the services'
+	// certificates of re-encrypt routes. When Routes don't specify their own
+	// destinationCACertificate, defaultDestinationCATrust is used.
+	//
+	// The configmap must contain the following key and data:
+	//
+	//   ca-bundle.crt: trusted CA certificates, including the OpenShift service CA.
+	//
+	// The user is responsible for including OpenShift service CA along with their own
+	// trusted CA certificates in the referenced configmap. OpenShift service CA can be
+	// found in openshift-service-ca.crt configmap in openshift-config namespace.
+	//
+	// +optional
+	DefaultDestinationCATrust *corev1.LocalObjectReference `json:"defaultDestinationCATrust,omitempty"`
```

Comment:
Define your own reference type (see the doc on corev1.ObjectReference), but otherwise the API field makes sense.  Since we know ClusterTrustBundles are coming, nest under a discriminated union

```go
DefaultDestinationTrust DefaultDestinationTrust

type DefaultDestinationTrust struct{
  // currently only "", ConfigMap, but later probably ClusterTrustBundle or SignerName
  Type string

  ConfigMap *corev1.LocalObjectReference `json:"defaultDestinationCATrust,omitempty"`
}
```